src/vs/workbench/contrib/notebook/browser/viewParts/notebookKernelActionViewItem.ts

```diff
@@ -64,7 +64,7 @@ export class NotebooKernelActionViewItem extends ActionViewItem {
 	private _updateActionFromKernelInfo(info: INotebookKernelMatchResult): void {
 
 		this._action.enabled = true;
-		const selectedOrSuggested = info.selected ?? info.suggestions[0];
+		const selectedOrSuggested = info.selected ?? (info.all.length === 1 && info.suggestions.length === 1 ? info.suggestions[0] : undefined);
```

Comment:
This was missing, users would get prompted to select a kernel but kernel picker would not read `Select Kernel` without this change.